code/Test_definitions/Geofencing.feature

```diff
@@ -0,0 +1,261 @@
+
+  @geofencing_subscriptions_11_Get_Event-Details_of_subscription_entered
+  Scenario: Subscription creation when service have area-entered event
+    Given  the appropriate values are used for geofencing
+    When the create subscription with event have area-entered at "Place1"
+    When the create subscription with event have area-entered at "Place2" and device enters "Place2"
+    Then the get event details from notifications-url
+    Then the response  code is 200
+    And the response header "Content-Type" is "application/json"
+    And the response header "x-correlator" has same value as the request header "x-correlator"
+    # The response has to comply with the generic response schema which is part of the spec
+    And the response body complies with the OAS schema at "/components/schemas/Subscription"
+
+  @geofencing_subscriptions_12_Get_Event-Details_of_subscription_left
+  Scenario: Subscription creation when service have area-left event
+    Given  the appropriate values are used for geofencing
+    When the create subscription with event have area-left with at "Place1"
+    When the create subscription with event have area-left with at "Place2" and device left "Place1"
+    Then they get event details from notifications-url
+    Then the response  code is 200
+    And the response header "Content-Type" is "application/json"
+    And the response header "x-correlator" has same value as the request header "x-correlator"
+    # The response has to comply with the generic response schema which is part of the spec
+    And the response body complies with the OAS schema at "/components/schemas/Subscription"
+
+  @geofencing_subscriptions_13_Getting_of_subscription_when_service_unavailable
+  Scenario: Getting Subscription when service is unavailable
+    Given  the appropriate values are used for geofencing
+    When the  get subscription method is triggered  when service is unavailable
+    Then the response  code is 503
+	And the response property "$.status" is 503
+    And the response property "$.code" is "UNAVAILABLE"
+    And the response property "$.message" contains a user friendly text
+
+
+  @geofencing_subscriptions_14_Deletion_of_subscription_when_service_unavailable
+  Scenario: Deletion of Subscription when service unavailable
+    Given  the appropriate values are used for geofencing
+    When the delete subscription method is triggered when service is unavailable
+    Then the response  code is 503
+	And the response property "$.status" is 503
+    And the response property "$.code" is "UNAVAILABLE"
+    And the response property "$.message" contains a user friendly text
+
+    
+  @geofencing_subscriptions_15_Get_invalid_geofencing_subscription_for_a_device
+  Scenario:  Get geofencing subscription with invalid subscription-id format
+    Given the appropriate values are used for geofencing
+    When the  get subscription method is triggered with invalid subscription-id which is in invalid format
```

Comment:
What is invalid format for a subscriptionId which is a string in the spec without any specific format?

Reply:
Invalid format may be empty or null string, or another data type like integer or bool.